common/src/main/java/org/apache/celeborn/common/write/PushState.java

```diff
@@ -40,25 +42,18 @@ public void cleanup() {
     inFlightRequestTracker.cleanup();
   }
 
-  // key: ${master addr}-${slave addr} value: list of data batch
-  public final ConcurrentHashMap<String, DataBatches> batchesMap = JavaUtils.newConcurrentHashMap();
-
-  /**
-   * Not thread-safe
-   *
-   * @param addressPair
-   * @param loc
-   * @param batchId
-   * @param body
-   * @return
-   */
-  public boolean addBatchData(String addressPair, PartitionLocation loc, int batchId, byte[] body) {
+  // key: ${master addr}, ${slave addr} value: list of data batch
+  public final ConcurrentHashMap<Pair<String, String>, DataBatches> batchesMap =
```

Comment:
use commons-lang3 Pair to replace String with delimiter `-`, to address hostname contains `-` case cc @waitinfuture